src/mlpack/core/optimizers/cne/cne.hpp

```diff
@@ -0,0 +1,234 @@
+/**
+ * @file cne.hpp
+ * @author Marcus Edel
+ * @author Kartik Nighania
+ *
+ * Conventional Neural Evolution
+ * An optimizer that works like biological evolution which selects best
+ * candidates based on their fitness scores and creates new generation by
+ * mutation and crossover of population.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CNE_CNE_HPP
+#define MLPACK_CORE_OPTIMIZERS_CNE_CNE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Conventional Neural Evolution (CNE) is a class of evolutionary algorithms
+ * focused on dealing with fixed topology. This class implements this algorithm
+ * as an optimization technique to converge a given function to minima.
+ *
+ * The algorithm works by creating a fixed number of candidates. The candidates
+ * are simply the network parameters that are untrained. Given the train and 
+ * test set. Each candidate is tested upon the training set or 1 epoch and then
+ * a fitness score is assigned to it. Given the selection percentage of best
+ * candidates by the user, for a single generation that many percentage of 
+ * candidates were selected for the next generation and the rest were removed
+ * The selected candidates for a particular generation then become the parent 
+ * for the next generation and evolution takes place.
+ *
+ * The evolution process basically takes place in two types-
+ * Crossover
+ * Mutation 
+ *
+ * Crossover takes two parents and generates two childs from them. Both the
+ * child have properties inherited by their parents. The parameters of the
+ * parents gets mixed using equal probability selection, creating two childs.
+ * This is just a mix of link weights or the parameters.
+ *
+ * In mutation parameters are updated by pertubating small noise.
+ * If $$ \Lambda $$ is the number of weights in the network.
```

Comment:
Have you tested the formulas renderd correctly? I guess you should use: ``\f[`` and ``\f$`` for more informatik take a look at: https://www.stack.nl/~dimitri/doxygen/manual/formulas.html